sdk/core/core-rest-pipeline/src/retryStrategies/retryStrategy.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureLogger } from "@azure/logger";
+import { PipelineResponse } from "../interfaces";
+import { RestError } from "../restError";
+
+/**
+ * State that keeps track of the last retry and controls how to do the next retries.
+ */
+export interface RetryStrategyState {
+  /**
+   * Maximum number of retries.
+   */
+  maxRetries?: number;
```

Comment:
why do we need `maxRetries` on state? It should remain the same.

Reply:
I didn’t want to put it in the strategy’s body or on the parameters to retryPolicy since this parameter generally it shouldn’t be needed. Having it in the state also allows dynamically changing the maximum number of retries, which isn’t relevant today but it could be in the future.

I can definitely imagine retry strategies that might know of a certain error case in which it could be beneficial to extend the configured number of retries, for example.

Reply:
With this update we don’t need maxRetries on the state anymore: https://github.com/Azure/azure-sdk-for-js/pull/19078/commits/706de41ce49045e4afaa8559dafcd8204a1e122c